README.md

```diff
@@ -168,305 +116,247 @@ Typical Java threads
   * "DestroyJavaVM"
   * "RMI TCP Accept-0"
   * "Service Thread"
-
-#### Jetty threads
-
-Jetty threads used for handling requests
-
+* **Jetty threads** - Jetty threads used for handling requests
   * "org.eclipse.jetty.server.session.HashSessionManager" {1..n}
   * "qtp715521683" {1.n}
 
-#### Emissary threads
-  * "FileInput-target/data/InputData" (standalone) - watches directory and processes files that show up
-  * "FileQueServer" (cluster) - communicated with the Feeder and gets files for processing
-  * "HeartbeatManager" (cluster) - tracks the state of the cluster
-  * "ResourceWatcher" - tracks the resource being consumed by Emissary/Place
-  * "MoveSpool" - this tracks the incoming data sent to Emissary
-  * "MobileAgent" {1..n} - these are the processing thread that work on the data
+### Base Data Object
+
+Data is sessionized and passed through Emissary in the form of a [BaseDataObject](src/main/java/emissary/core/BaseDataObject.java).
 
 ### Processing Flow
 
-There are various adapters to abstract where data comes from.  These sources ultimately end up in PickUpPlace broken out
- sessions.  PickUpPlace then pulls a MobileAgent from the thread pool and hands off the session for processing.
+There are various adapters to abstract how data is picked up by Emissary.  Ultimately the data is received in PickUpPlace and is broken out
+ into BaseDataObject sessions.  PickUpPlace then pulls a MobileAgent from the thread pool and hands off the session for processing in [assignToPooledAgent()](src/main/java/emissary/pickup/PickUpPlace.java#L719).
+
+### Processing Routing
+
+MobileAgent walks a given session through all the relevant places.
+
+There are two types of MobileAgent:
+* [MobileAgent](src/main/java/emissary/core/MobileAgent.java) - Used for processing a single payload
+* [HDMobileAgent](src/main/java/emissary/core/HDMobileAgent.java) - Used for processing a bundle of payloads
+
+Processing starts when PickUpPlace's assignToPooledAgent() calls [agent.go(payload, startingLocation)](src/main/java/emissary/pickup/PickUpPlace.java#L751)
 
-#### Processing Routing
+The MobileAgent processing loop is handled in [agentControl()](src/main/java/emissary/core/MobileAgent.java#L249)
```

Comment:
I would not put line-numbers in here as that could be brittle.

Reply:
I hear what you are saying. I do think it is helpful to have links to the functions so that devs can easily see the code referenced.

Based on this stackoverflow article it looks like you can reference a specific commit, then the line numbers wouldn't change: https://stackoverflow.com/questions/18220650/github-link-to-function-in-source

Would that be a good compromise for us? Then we know the line numbers will be valid and the dev can still easily link to the function. The only down-side is that the file they are viewing may not reflect the current code